lib/archethic/self_repair/notifier.ex

```diff
@@ -105,7 +119,129 @@ defmodule Archethic.SelfRepair.Notifier do
+          {summary_time :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())}
+  def summary_aggregates_to_sync(summary_time) do
+    node_list =
+      P2P.list_nodes()
+      |> Enum.filter(fn node ->
+        node.authorization_date != nil &&
+          DateTime.compare(node.authorization_date, summary_time) == :lt
+      end)
+
+    previous_chain_storage_nodes =
+      summary_time
+      |> Crypto.derive_beacon_aggregate_address()
+      |> Election.chain_storage_nodes(node_list)
+
+    {summary_time, previous_chain_storage_nodes}
+  end
+
+  @spec sync_unavailable_summary_aggregate(
+          {summary_time :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())},
+          current_node_public_key :: Crypto.key(),
+          unavailable_node_key :: Crypto.key()
+        ) :: :ok
+  def sync_unavailable_summary_aggregate(
+        {summary_time, previous_chain_storage_nodes},
+        current_node_public_key,
+        unavailable_node_key
+      ) do
+    new_chain_storage_nodes =
+      Election.chain_storage_nodes(
+        Crypto.derive_beacon_aggregate_address(summary_time),
+        P2P.authorized_nodes()
+        |> Enum.reject(&(&1.first_public_key == unavailable_node_key))
+      ) -- previous_chain_storage_nodes
+
+    with {:empty, false} <- {:empty, Enum.empty?(new_chain_storage_nodes)},
+         #  current node should not be part of previous_chain_storage_nodes as it would already have aggregate
+         {:prev_election, false} <-
+           {:prev_election,
+            Utils.key_in_node_list?(previous_chain_storage_nodes, current_node_public_key)},
+         # current node should be part of new_chain storage nodes to do the pull
+         {:new_election, true} <-
+           {:new_election,
+            Utils.key_in_node_list?(new_chain_storage_nodes, current_node_public_key)},
+         # filter out unavailable nodes from prev strage nodes to fetch aggregate
+         storage_nodes when storage_nodes != [] <-
+           Enum.filter(previous_chain_storage_nodes, &(&1.available? == true)),
+         #  quorum read for aggregate fetching
+         {:ok, aggregate = %SummaryAggregate{}} <-
+           BeaconChain.fetch_summaries_aggregate_from_nodes(summary_time, storage_nodes) do
+      # Pull mechnism to fetch summary aggreagete from local calulation os shards
+      BeaconChain.write_summaries_aggregate(aggregate)
+
+      Logger.debug(
+        "Fetched and Stored Missing #{summary_time} Aggregate due to shard unavailablity",
+        Repair: "SelfRepair.Notifier"
+      )
+
+      :ok
+    else
+      {:empty, true} ->
+        Logger.debug("AggregateRepair: #{summary_time} : omitted ",
+          reason:
+            "There are currently no new nodes in the New (Election)/(shard selection) for Chain Storage Nodes. and Summarytime"
+        )
+
+      {:prev_election, true} ->
+        Logger.debug("AggregateRepair: #{summary_time} : omitted ",
```

Comment:
It will be better to concatenate the reason in the message instead of using metadata.